### DISCUSSION TITLE
How to add comment in a visitor

### DISCUSSION THREAD

**Discussion by SyMind** [swc-project/swc#7349] [General]
I'm writing a project based on SWC. I want to add some comments in a visitor when transforming AST, like below:

input:
```jsx
import Svg from 'react-native-svg';
<svg><g/><div/></svg>;
```

output:
```jsx
import Svg, { G } from 'react-native-svg';
/* SVGR has dropped some elements not supported by react-native-svg: div */
<Svg><G/></Svg>;
```

**Comment by kwonoj** [comment]
did `add_leading_comments` not work?

**Reply by SyMind** [reply]
`add_leading_comments ` seems like private function. I cannot call it in my Visitor. Can you give me some examples?

**Reply by kwonoj** [reply]
it is public trait. https://github.com/swc-project/swc/blob/e9f3371c5ff029555ead23f10039b02830e86c64/crates/swc_common/src/comments.rs#LL33C1-L33C1
